verify/code/tutorials/cursor.md

```diff
@@ -0,0 +1,253 @@
+  const userCredential = await createUserWithEmailAndPassword(auth, email, password);
+}
+```
+
+---
+
+### Architecture regression
+
+**What happened**: AI reverted Firebase authentication back to localStorage approach.
+
+**AI generated**:
+```typescript
+// src/contexts/AuthContext.tsx - Correct Firebase approach
+const login = async (email: string, password: string) => {
+  const userCredential = await signInWithEmailAndPassword(auth, email, password);
+  return !!userCredential.user;
+};
+```
+
+```typescript
+// AI reverted to original localStorage approach
+const [user, setUser] = useState(() => {
+  const savedUser = localStorage.getItem('user'); // ❌ Back to localStorage!
+  return savedUser ? JSON.parse(savedUser) : null;
+});
+
+const login = async (email: string, password: string) => {
+  const response = await fetch('/api/auth/login', { // ❌ API calls instead of Firebase
+    method: 'POST',
+    body: JSON.stringify({ email, password })
+  });
+};
+```
+
+**Verify Code caught**:
+
+![Verify Code alert showing architecture regression from Firebase back to localStorage](/images/verify/code/tutorials/cursor/tutorial-cursor-3.webp){ width="75%" }
+
+**P1 - Intent (High)**: AI reverted Firebase authentication implementation back to using localStorage and API calls.
+
+**Why this matters**: Lost all Firebase benefits like real-time auth state, secure token management, and cross-device sessions. Back to the original problems we were trying to solve.
+
+**Correct approach**:
+```typescript
+// src/contexts/AuthContext.tsx
+const login = async (email: string, password: string) => {
+  const userCredential = await signInWithEmailAndPassword(auth, email, password);
+  return !!userCredential.user;
+};
+```
+
+---
+
+## The results
+
+Verify Code caught **four critical issues** across a "simple" five-step plan:
+
+1. **Incomplete implementation** - Step one missed core functionality.
+2. **Breaking changes** - Step four deleted working code.
+3. **Security vulnerabilities** - Step five ignored input validation.
+4. **Architecture regression** - Step three went backwards.
+
+### Successful implementation achieved
+
+By following Verify Code's guidance at each step, Gemini 2.5 Flash successfully completed the Firebase migration. Users can now register and authenticate properly.
+
+**Firebase console showing the code@verify.com user creation:**
+
+![Firebase Authentication console showing successfully created users](/images/verify/code/tutorials/cursor/tutorial-cursor-4.webp)
+
+**Successful login in the e-commerce app and Firebase user created:**
+
+![E-commerce app showing successful login with code@verify.com user](/images/verify/code/tutorials/cursor/tutorial-cursor-5.webp)
+
+The migration from localStorage to Firebase authentication was completed without the typical debugging cycles. [Verify Code](/verify/code/) caught each issue in real-time, allowing us to fix problems immediately rather than discovering them during testing.
```

Comment:
```suggestion
The migration from `localStorage` to Firebase authentication was completed without the typical debugging cycles. [Verify Code](/verify/code/) caught each issue in real-time, allowing us to fix problems immediately rather than discovering them during testing.
```